tests/core/backslashes-act-literally.js.test

```diff
@@ -0,0 +1,145 @@
+
```

Comment:
OK I realize I'm u-turning here, but this seems not super useful...

Would be much easier to test this if you just compiled it and looked at the resulting action.
With this test, refactoring JS generation now is going to require rewriting these tests entirely, and it feels like we shouldn't care how it compiles, just that the result i the same. WDYT?

Reply:
I disagree, I think there is definite value here.

For the same reason that it was valuable to have the original tests in `lexer.spec.ts` - because it's hard to make sure that we properly escape all the relevant escape characters, handle special text in strings properly, etc etc - these test cases are very valuable, because they run before any transpiled code is actually run.

Otherwise, when trying to track down - for example - how we are mishandling `\` characters, it's very difficult to actually run any tests, because compilation (to a `CompiledGraph`) just fails, with invalid transpiled JavaScript and nonsense output.